internal/util/function/rerank/function_score.go

```diff
@@ -20,38 +20,79 @@ package rerank
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
+	"reflect"
+	"strconv"
 	"strings"
 
 	"github.com/samber/lo"
 
+	"github.com/milvus-io/milvus-proto/go-api/v2/commonpb"
 	"github.com/milvus-io/milvus-proto/go-api/v2/milvuspb"
 	"github.com/milvus-io/milvus-proto/go-api/v2/schemapb"
+	"github.com/milvus-io/milvus/pkg/v2/util/funcutil"
 	"github.com/milvus-io/milvus/pkg/v2/util/merr"
 )
 
 const (
 	decayFunctionName string = "decay"
 	modelFunctionName string = "model"
+	rrfName           string = "rrf"
+	weightedName      string = "weighted"
 )
 
+const (
+	maxScorer string = "max"
+	sumScorer string = "sum"
+	avgScorer string = "avg"
+)
+
+// legacy rrf/weighted rerank configs
+
+const (
+	legacyRankTypeKey   = "strategy"
+	legacyRankParamsKey = "params"
+)
+
+type rankType int
+
+const (
+	invalidRankType  rankType = iota // invalidRankType   = 0
+	rrfRankType                      // rrfRankType = 1
+	weightedRankType                 // weightedRankType = 2
+)
+
+var rankTypeMap = map[string]rankType{
+	"invalid":  invalidRankType,
+	"rrf":      rrfRankType,
+	"weighted": weightedRankType,
+}
+
 type SearchParams struct {
 	nq           int64
 	limit        int64
 	offset       int64
 	roundDecimal int64
 
-	// TODO: supports group search
 	groupByFieldId  int64
 	groupSize       int64
 	strictGroupSize bool
+	groupScore      string
 
 	searchMetrics []string
 }
 
-func NewSearchParams(nq, limit, offset, roundDecimal, groupByFieldId, groupSize int64, strictGroupSize bool, searchMetrics []string) *SearchParams {
+func (s *SearchParams) isGrouping() bool {
+	return s.groupByFieldId > 0
+}
+
+func NewSearchParams(nq, limit, offset, roundDecimal, groupByFieldId, groupSize int64, strictGroupSize bool, groupScore string, searchMetrics []string) *SearchParams {
+	if groupScore == "" {
+		groupScore = maxScorer
```

Comment:
I saw similar things in the caller of this function. Let's remove them

Reply:
fixed